### PR TITLE
get by user id api/quiz-results modify

### PR DESCRIPTION
**Pull request by qamarfj**
# Description

I have modify the api\controllers\quiz-results.controller.js . Now  it returns latest records from quiz_results with given user id up to the total number of questions in questions table. 
Fixes # (issue) backend part of issue
#136 

# How to test?
try to get results with userId from Swagger at following address it will return latest records up to max number of questions in questions table. 
http://localhost:5000/api/documentation/#/Quiz%20Results/get_quiz_results

# Checklist

- [x ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/controllers/quiz-results.controller.js [hunk 1]
```diff
@@ -24,9 +24,15 @@ const getQuizResults = async (answerId, userId) => {
         .where('fk_answer_id', '=', answerId)
         .select('id', 'fk_answer_id', 'fk_user_id', 'created_date');
     } else if (userId !== undefined) {
+      // get total number of questions from questions table.
+      const numberOfQuestions = (await knex('questions').count())[0][
+        'count(*)'
+      ];
```
Comment (bytewiz) [inline src/server/api/controllers/quiz-results.controller.js:30]: This is a bit hard to read :) and does this work? and could there maybe be a different way of writing this?

File: src/server/api/controllers/quiz-results.controller.js [hunk 2]
```diff
@@ -24,9 +24,15 @@ const getQuizResults = async (answerId, userId) => {
...
+      // Get latest records with userid from quiz_results upto to numberOfQuestions
       filteredResults = await knex('quiz_results')
         .where('fk_user_id', '=', userId)
-        .select('id', 'fk_answer_id', 'fk_user_id', 'created_date');
+        .orderBy('created_date', 'desc')
+        .limit(Number(numberOfQuestions));
```
Comment (bytewiz) [inline src/server/api/controllers/quiz-results.controller.js:35]: why are you using `Number()`?

File: src/server/api/controllers/quiz-results.controller.js (hunk 1 above)
Comment (bytewiz) [reply src/server/api/controllers/quiz-results.controller.js:30]: So doesn't this return the same?
```suggestion
      const numberOfQuestions = await knex.raw('select count(*) from questions')
```

Comment (qamarfj) [reply src/server/api/controllers/quiz-results.controller.js:30]: no it does not. it returns complicated object and then we have to pick  from that object.

Comment (qamarfj) [reply src/server/api/controllers/quiz-results.controller.js:30]: [](url)
vlcscreen

**Review by dpfernandes — APPROVED** [review]
👍

**Review by TITHI1244 — APPROVED** [review]
Works perfectly as it is supposed to be. Nice work @qamarfj

**Pull request merged**